From 5169d43ce8085726d1d9945fef636154a21eaf17 Mon Sep 17 00:00:00 2001
From: Ovidiu Poncea <ovidiu.poncea@windriver.com>
Date: Thu, 20 Dec 2018 07:18:55 -0500
Subject: [PATCH 5/5] Add StarlingX specific restart command for Ceph monitors

Since we don't use systemd to manage Ceph and we have pmon monitoring we
have to make sure that:
1. Restarting is properly handled as "systemctl restart" will return error
   and manifest will fail;
2. Pmon does not check ceph-mon status during restart. Otherwise we risk
   getting into a race condition between the puppet restart and pmon
   detecting that ceph is down and trying a restart.

Both are resolved when using /etc/init.d/ceph-init-wrapper restart

Signed-off-by: Ovidiu Poncea <Ovidiu.Poncea@windriver.com>
---
 manifests/mon.pp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/manifests/mon.pp b/manifests/mon.pp
index 7cbec0d..3936529 100644
--- a/manifests/mon.pp
+++ b/manifests/mon.pp
@@ -106,6 +106,7 @@ define ceph::mon (
         start    => "service ceph start mon.${id}",
         stop     => "service ceph stop mon.${id}",
         status   => "service ceph status mon.${id}",
+        restart  => "/etc/init.d/ceph-init-wrapper restart mon.${id}",
         enable   => $mon_enable,
       }
     }
-- 
2.30.2

